apps/ui/src/components/FormAuctionBid.vue

```diff
@@ -0,0 +1,263 @@
+import { AuctionDetailFragment } from '@/helpers/auction/gql/graphql';
+import { CHAIN_IDS } from '@/helpers/constants';
+import { getProvider } from '@/helpers/provider';
+import { _n, _t } from '@/helpers/utils';
+
+const DEFAULT_PRICE_PREMIUM = 1.001; // 0.1% above clearing price
+const PRICE_DECIMALS = 4;
+const INVERTED_PRICE_DECIMALS = 8;
+
+const INPUT_DEFINITION = {
+  type: 'string',
+  examples: ['0.0']
+};
+
+const props = defineProps<{
+  auction: AuctionDetailFragment;
+  network: AuctionNetworkId;
+}>();
+
+const { web3Account } = useWeb3();
+const { modalAccountOpen } = useModal();
+
+const bidAmount = ref('');
+const bidPrice = ref('');
+const isPriceInverted = ref(false);
+
+const provider = computed(() => getProvider(Number(CHAIN_IDS[props.network])));
+
+const { data: userBalance, isError: isBalanceError } = useQuery({
+  queryKey: ['balance', web3Account, () => props.auction.addressBiddingToken],
+  queryFn: async () => {
+    if (!web3Account.value) return '0';
+
+    const contract = new Contract(
+      props.auction.addressBiddingToken,
+      abis.erc20,
+      provider.value
+    );
+
+    const balance = await contract.balanceOf(web3Account.value);
+
+    return balance.toString();
+  },
+  enabled: computed(() => !!web3Account.value)
+});
+
+const { data: totalSupply, isError: isSupplyError } = useQuery({
+  queryKey: ['supply', () => props.auction.addressAuctioningToken],
+  queryFn: async () => {
+    const contract = new Contract(
+      props.auction.addressAuctioningToken,
+      abis.erc20,
+      provider.value
+    );
+
+    const totalSupply = await contract.totalSupply();
+
+    return totalSupply.toString();
+  }
+});
+
+const formattedBalance = computed(() => {
+  if (!userBalance.value) return 0;
+  return parseFloat(
+    formatUnits(userBalance.value, props.auction.decimalsBiddingToken)
+  );
+});
+
+const hasBalance = computed(() => !!(web3Account.value && userBalance.value));
+
+const priceLabel = computed(() =>
+  isPriceInverted.value
+    ? `${props.auction.symbolAuctioningToken} per ${props.auction.symbolBiddingToken}`
+    : `${props.auction.symbolBiddingToken} per ${props.auction.symbolAuctioningToken}`
+);
+
+const maxMarketCap = computed(() => {
+  const displayPrice = parseFloat(bidPrice.value) || 0;
+  if (!displayPrice || !totalSupply.value) return '0';
+
+  const totalSupplyFormatted = parseFloat(
+    formatUnits(totalSupply.value, props.auction.decimalsAuctioningToken)
+  );
+  const pricePerToken = isPriceInverted.value ? 1 / displayPrice : displayPrice;
+
+  const precision = isPriceInverted.value
+    ? INVERTED_PRICE_DECIMALS
+    : PRICE_DECIMALS;
+
+  const marketCapValue = Math.floor(
+    totalSupplyFormatted * +pricePerToken.toFixed(precision)
+  );
+
+  return _n(marketCapValue);
+});
+
+const amountError = computed(() => {
+  if (!bidAmount.value) return undefined;
+
+  const amount = parseFloat(bidAmount.value);
+  if (amount <= 0) return 'Invalid amount';
```

Comment:
Is there not a existing validation that we could use? If we use validation logic we already have we could just have it handle format, minimum and maximum amounts etc instead of reinventing it there.